### ISSUE TITLE
Memory.getBytes returns empty array

### ISSUE THREAD

**Issue by tombonner** [justfoxing/ghidra_bridge#52]
Hi,

I am trying to read memory using the following:

```
buffer = array.array ("b", b"\0" * length)
currentProgram.getMemory().getBytes(currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(hex(start)), buffer)
```

When I try to read say, 16 bytes length from 0x400000 (expecting to see an MZ header), I get:

array('b', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])

However, if I jiggle the code about to only read a single byte I get:

```
buffer = currentProgram.getMemory().getByte(currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(hex(start)))
print(buffer)
77
```

Am I doing something stupid, or is there an issue with the array handling or something?!

Regards,

Tom

**Comment by justfoxing** [comment]
Hi, Tom.

Could you give me the actual code you're running across the GhidraBridge to get this issue? I'm unable to replicate it across the bridge as written (using ghidra_bridge 0.2.5, jfx_bridge 0.7.1, and Ghidra 9.1.2) - when I make some modifications to work around an unrelated issue (thanks for helping me find that! [but it's not your problem :P]), I get the expected behaviour (data populated in the buffer - I'm using an ELF file instead of a PE because that's what I had open, but that won't make a difference). Here's what works for me:

```python
In [1]: import ghidra_bridge
In [2]: b = ghidra_bridge.GhidraBridge(namespace=globals())
In [3]: array = b.remote_import("array")
In [4]: start = 0x100000
In [5]: length = 16
In [6]: buffer = b.remote_eval("array.array('b', data)", array=array, data = b"\0"*length) # need to remote_eval this to work around "b" being set to unicode by the bridge
In [7]: print(buffer)
array('b', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
In [8]: currentProgram.getMemory().getBytes(currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(hex(start)), buffer)
Out[8]: 16
In [9]: print(buffer)
array('b', [127, 69, 76, 70, 2, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```
Similarly, when I run your code directly in the Ghidra python interpreter, I get the expected behaviour:

```python
>>> import array
>>> length = 16
>>> start = 0x100000
>>> buffer = array.array ("b", b"\0" * length)
>>> print(buffer)
array('b', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
>>> currentProgram.getMemory().getBytes(currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(hex(start)), buffer)
16
>>> print(buffer)
array('b', [127, 69, 76, 70, 2, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

**Comment by justfoxing** [comment]
Added a fix to the unrelated issue in jfx_bridge 0.8.0. So with that, the bridged code now reads:

```python
In [1]: import ghidra_bridge
In [2]: b = ghidra_bridge.GhidraBridge(namespace=globals())
In [3]: array = b.remote_import("array")
In [4]: start = 0x100000
In [5]: length = 16
In [6]: buffer = array.array("b", b"\0"*length)
In [7]: print(buffer)
array('b', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
In [8]: currentProgram.getMemory().getBytes(currentProgram.getAddressFactory().getDefaultAddressSpace().getAddress(hex(start)), buffer)
Out[8]: 16
In [9]: print(buffer)
array('b', [127, 69, 76, 70, 2, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

It makes no difference to your problem, just slightly cleaner.

**Comment by tombonner** [comment]
Ahh, so I was doing something stupid!

This was key...

```
array = self.bridge.remote_import("array")
buffer = self.bridge.remote_eval("array.array('b', data)", array=array, data = b"\0"*length) 
```

With that in place everything works very much as expected, thanks for your assistance!
